src/main/java/org/htmlunit/javascript/host/URLSearchParams.java

```diff
@@ -137,26 +143,171 @@ public URLSearchParams() {
+                Object value = nameValueIterator.hasNext() ? nameValueIterator.next() : null;
+
+                if (name == null || value == null || nameValueIterator.hasNext()) {
+                    throw ScriptRuntime.typeError("Sequence initializer must only contain pair elements.");
+                }
+
+                if (sb.length() > 0) {
+                    sb.append('&');
+                }
+
+                sb.append(Context.toString(name)).append('=').append(Context.toString(value));
+            }
+
+            return sb.toString();
+        } catch (EcmaError e) {
+            throw ScriptRuntime.typeError("Failed to construct 'URLSearchParams': " + e.getErrorMessage());
+        }
+    }
+
+    /*
+     * Java iterator for JavaScript objects implementing the iterator protocol:
+     * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols
+     */
+    private static class IterationProtocolIterator implements java.util.Iterator<Object> {
+        private static final Object UNSET = new Object();
+        private static final Object NO_MORE = new Object();
+
+        private final Context cx;
+        private final Scriptable scope;
+
+        private final Scriptable iterator;
+        private final Function nextFunc;
+
+        private Object nextValue = UNSET;
+
+        public IterationProtocolIterator(final Scriptable iterable, final Context cx, final Scriptable scope) {
+            this.cx = Objects.requireNonNull(cx);
+            this.scope = Objects.requireNonNull(scope);
+
+            // get iterator function
+            Object iteratorFunc = ScriptableObject.getProperty(iterable, SymbolKey.ITERATOR);
+            if (iteratorFunc == Undefined.instance) {
+                throw ScriptRuntime.typeError("[Symbol.iterator] is not a function");
+            }
+            if (!(iteratorFunc instanceof Function)) {
+                throw ScriptRuntime.typeError("@@iterator must be a callable");
+            }
+
+            // get iterator object by calling iterator function
+            Object iterator = ((Function)iteratorFunc).call(cx, scope, iterable, new Object[0]);
+            if (iterator == Undefined.instance) {
+                throw ScriptRuntime.typeError("[Symbol.iterator]() is undefined");
+            }
+            if (!(iterator instanceof Scriptable)) {
+                throw ScriptRuntime.typeError("Iterator object must be an object.");
+            }
+            this.iterator = (Scriptable)iterator;
+
+            // iterator object must always implement next()
+            Object nextFunc = ScriptRuntime.getObjectProp(iterator, "next", cx, scope);
+            if (nextFunc == Undefined.instance) {
+                throw ScriptRuntime.typeError("Expected next() function on iterator.");
+            }
+            if (!(nextFunc instanceof Function)) {
+                throw ScriptRuntime.typeError("iterator.next is not a function");
+            }
```

Comment:
```suggestion
            if (!(nextFunc instanceof Function)) {
                throw ScriptRuntime.typeError("Expected next() function on iterator.");
            }
```

`new URLSearchParams({[Symbol.iterator]: function (){ return {}; }})` -> `Expected next() function on iterator.`